course_management.lua

```diff
@@ -170,55 +172,43 @@ function courseplay:loadCourse(vehicle, id, useRealId, addCourseAtEnd) -- fn is
 				end;
 				courseplay:debug(string.format('course 1 has %d crossing points (excluding first point), course 2 has %d crossing points (excluding last point), useFirstMatch=%s', #crossingPoints[1], #crossingPoints[2], tostring(useFirstMatch)), 8);
 
-				--find < 50m match
-				local smallestDist = math.huge;
+				--find < wpDistMax match with lowest total turn angle
+				local smallestAngle = math.huge;
 				if #crossingPoints[1] > 0 and #crossingPoints[2] > 0 then
 					for _,wpNum1 in pairs(crossingPoints[1]) do
 						local wp1 = course1[wpNum1];
 						for _,wpNum2 in pairs(crossingPoints[2]) do
 							local wp2 = course2[wpNum2];
 							local dist = courseplay:distance(wp1.cx, wp1.cz, wp2.cx, wp2.cz);
+							--Calculate actual turn direction between pair of crosspoints
+							local angleTurn = 180 * math.atan2(wp2.cx-wp1.cx,wp2.cz-wp1.cz) / math.pi -- convert to degrees  
```

Comment:
Use math.deg() to convert to degrees.

(BTW, with your angleDiff with degrees you still have to convert one of the parameters, just like with the existing function, so don't see the value. Also, the Giants engine works with radians as well.)

Reply:
Thanks, will do math.deg()

The reason I try to stick to degrees as much as possible is because waypoints are stored in XML in degrees, but more importantly - we are about to hard-code some values (45, 90 degrees) to compare angles and make decisions. 

I am not good at visualizing radians without first converting them to degrees, so I feel better when I see IF operations be done with degrees.

That said the angleDiff function is called only once, so I think I can replace that expression with bunch of math convert calls and use existing radian procedure as you suggest. Your call. 

I did notice that there is a lot of redundant functions in various parts of the code, so I am guilty of thinking "whats wrong with one more?" :-).

Given how bad I am with knowing even lua math library I do like to apologize for my future behavior of not taking the time to hunt down existing functions in rather obscure (to me) locations such as geo.lua. I do promise to fix them if you pinpoint them to me though...

Reply:
No need to apologize, we appreciate your contribution. We are aware that the CP code isn't in the best shape and are in the process of trying to change that.

Reply:
if you change the trigger value, you should change the value in line 57 as well to 96 or so....

Reply:
Aaaa. How do I delete this pull request? I did create a better one. 
Had to delete the fork for _this_ request. I bet you guys all work with Git command line and do not have any issues that I am experiencing with Github web interface...